shoes-core/lib/shoes/image.rb

```diff
@@ -21,9 +23,31 @@ def path
       @style[:file_path]
     end
 
-    def path=(path)
-      style(file_path: path)
+    def path=(path_or_data)
+      style(file_path: normalized_source(path_or_data))
       @gui.update_image
     end
+
+    def url?(path_or_data)
+      path_or_data =~ %r{^https?://}
+    end
+
+    def raw_image_data?(name_or_data)
+      name_or_data.encoding == BINARY_ENCODING
+    end
+
+    private
+
+    def normalized_source(path_or_data)
+      return path_or_data if raw_image_data?(path_or_data)
+      return path_or_data if url?(path_or_data)
+      absolute_file_path(path_or_data)
+    end
+
+    def absolute_file_path(path)
+      path = File.join(Shoes::DIR, path) unless Pathname(path).absolute?
+      fail FileNotFoundError, "#{path} not found." unless File.exist?(path)
+      path
+    end
```

Comment:
I believe some of the above methods are copied from the SWT/backend side of things or semi copied, right? If so it'd be nice to remove them from there and use those methods here or another way to avoid duplication